openlibrary/plugins/openlibrary/js/modals/index.js

```diff
@@ -392,6 +421,35 @@ function displayModal(content, maxWidth) {
     });
 }
 
+function displayDynamicModal(content, maxWidth) {
+    const maxWidthNum = parseInt(maxWidth.replace('px', ''));
+    const modalId = `#${content.id}`;
+    const context = content.dataset['context'] ? JSON.parse(content.dataset['context']) : null;
+    const reloadId = context ? context.reloadId : null;
+
+    function openModal() {
+        $.colorbox({
+            inline: true,
+            opacity: '0.5',
+            href: modalId,
+            width: '100%',
+            maxWidth: window.innerWidth > maxWidthNum ? maxWidth : '100%',
+            onClosed: function () {
+                if (reloadId) {
+                    $(`#${reloadId}`).trigger('contentReload');
+                }
+            },
+        });
+    }
+
+    openModal();
+
+    $(window).on('resize', function () {
+        $.colorbox.remove();
+        openModal();
+    });
+}
```

Comment:
It looks like this is exactly the same as the `displayModal` function, except for the change that sets `maxWidth` to `100%` if the screen's width is less than the value that is passed.

I think that we can avoid this repeated code by modifying `displayModal` to include the changes that you added to `displayDynamicModal`.

Reply:
Actually, this change may not be needed at all.  While testing, I replaced your `displayDynamicModal` call with a `displayModal` call, and everything looked the same in both mobile and desktop views (tested in Firefox and Chrome).

Reply:
Yes testing this myself and realizing displayDynamicModal is not needed. I can use the pre-existing `addClickListeners` and `addShareModalButtonListeners`. I was trying to make it resize appropriately with the dynamic version (eg. if you open in mobile and then switch to desktop, the modal gets out of proportion), but the dynamic isn't handling that correctly either. Is that something we need to take into account for modal testing? 
<img width="1263" alt="Screenshot 2024-05-31 at 11 52 16 AM" src="https://github.com/internetarchive/openlibrary/assets/79802377/72756c5f-8ced-4775-91f4-c3c4ef64fdb3">

Reply:
I doubt that this will affect many people, but thanks for pointing it out!